src/wp-includes/script-loader.php

```diff
@@ -2267,38 +2267,51 @@ function wp_enqueue_global_styles() {
 		return;
 	}
 
+	$separate_assets = wp_should_load_separate_core_block_assets();
+
+	/*
+	 * Global styles should be printed in the head when loading all styles combined.
+	 * The footer should only be used to print global styles for classic themes with separate core assets enabled.
+	 *
+	 * See https://core.trac.wordpress.org/ticket/53494.
+	 */
+	if ( ! $separate_assets && doing_action( 'wp_footer' ) || $separate_assets && doing_action( 'wp_enqueue_scripts' ) ) {
```

Comment:
This code is correct as per the operator precedence: ` &&` has higher precedence than `||`. Readers may benefit from grouping them with parenthesis so the two code paths are clearer. This is a style choice, though.

Reply:
Clarity is always preferred! Grouping now.